### PR TITLE
Minor: improve TableReference docs

### PR DESCRIPTION
**Pull request by alamb**
## Rationale for this change

While reviewing https://github.com/apache/arrow-datafusion/pull/9933 from @comphead  I noticed a few things that could be improved in the docs

## What changes are included in this PR?

Improve `TableReferenc` docs

## Are these changes tested?

Doc tests

## Are there any user-facing changes?
Better docs

### CODE REVIEW

**Review by comphead — APPROVED** [review]
lgtm thanks @alamb

File: datafusion/common/src/table_reference.rs [hunk 1]
```diff
@@ -139,7 +142,7 @@ impl TableReference {
 
     /// Convenience method for creating a [`TableReference::Partial`].
     ///
-    /// As described on [`TableReference`] this does *NO* parsing at all.
+    /// As described on [`TableReference`] this does *NO* normalization at all.
```
Comment (Jefffrey) [inline datafusion/common/src/table_reference.rs:145]: nit: technically the doc on `TableReference` doesn't make reference to `partial` or `full` functions as it only refers to `bare`

**Review by Jefffrey — APPROVED** [review]
:+1:

File: datafusion/common/src/table_reference.rs (hunk 1 above)
Comment (alamb) [reply datafusion/common/src/table_reference.rs]: That is a good call -- I have clarified in a1d5965fff9767f6d44088e547275e69edaa81bc. Thank you @Jefffrey

**Comment by alamb** [comment]
Thanks again @comphead  and @Jefffrey

**Pull request merged**